src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1229,125 @@ private int SkipBlockComment(int i)
             return i;
         }
 
-        private static char Backtick(char c)
+        private string Backtick(char c)
```

Comment:
I'm fine with `Backtick` returning a string, I think that's a reasonable change.

I'm less happy with the inconsistent handling at each call site - maybe it's necessary but it's worth discussing more.

It looks like the only reason to check the length is to support doubling `{` and `}`. Should that happen if someone wrote `` "`u7b"`` instead of `"{"`?